src/languages/php/tuning.md

```diff
@@ -0,0 +1,110 @@
+# Performance tuning PHP
+
+Once your application is up and running it still needs to be kept fast.  Platform.sh offers a wide degree of flexibility in how PHP behaves, but that does mean you may need to take a few steps to ensure your site is running optimally.
+
+The following recommendations are guidelines only.  They're also listed in approximately the order we recommend investigating them, although your mileage may vary.
+
+## Upgrade to PHP 7.1+
+
+There is very little purpose to trying to optimize a PHP application on PHP 5.  PHP 7 is generally twice as fast and uses half as much memory as PHP 5, making it unquestionably the first step to take when trying to make a PHP-based site run faster.
+
+To change your PHP version, simply change the `type` key in your `.platform.app.yaml` to the desired PHP version.  As always, test it on a branch first before merging to `master`.
+
+## Ensure that the router cache is properly configured
+
+Although not PHP-specific, a common source of performance issues is a misconfigured cache.  The most common issue is not whitelisting session cookies, which results in a site with any cookies at all, including from analytics tools, never being cached.  See the [router cache](/configuration/routes/cache.md) documentation, and the [cookie entry](/configuration/routes/cache.md#cookies) specifically.
+
+You will also need to ensure that your application is sending the correct `cache-control` header.  The router cache will obey whatever cache headers your application sends, so send it good ones.
+
+## Optimize the FPM worker count
+
+PHP-FPM reserves a fixed number of simultaneous worker processes to handle incoming requests.  If more simultaneous requests are received than the number of workers then some requests will wait.  The default worker count is deliberately set rather conservative but can be improved in many cases.  See the [PHP-FPM sizing](/languages/php/fpm.md) page for how to determine and set a more optimal value.
+
+If your `/var/log/app.log` file contains a lot of entries like 
+
+```
+WARNING: [pool web] server reached max_children setting (2), consider raising it
+```
+
+That means your worker count is too low.  If you have already set an optimal worker size and still see that message frequently consider upgrading your project resources.
```

Comment:
The common case is not a lack of resources, but requests being blocked on an external resource or something similar. Asking to upgrade the plan first is probably disingenuous.

Reply:
Is it?  I've not seen external resource-dependencies causing a max_children issue.  (And telling people to upgrade is what we already do in about 15 other places in the docs.)

Reply:
```I've not seen external resource-dependencies causing a max_children issue.```` I have - but it's rare. Most of the time the request is just not very well cached and is taking a bunch of seconds to reply, and has a bit of concurrency.